content/blog/2018-04-18-version-2.md

```diff
@@ -0,0 +1,220 @@
+
+
+## Computed properties
+
+Another thing that people often found confusing about Svelte is the way computed properties work. To recap, if you had a component with this...
+
+```js
+export default {
+	computed: {
+		d: (a, b, c) => a = b + c
+	}
+};
+```
+
+...then Svelte would first look at the function arguments to see which values `d` depended on, and then it would write code that updated `d` whenever those values changed, by injecting them into the function. That's cool, because it allows you to derive complex values from your component's inputs without worrying about when they need to recomputed, but it's also... *weird*. JavaScript doesn't work that way!
+
+In v2, we use [destructuring](http://www.jstips.co/en/javascript/use-destructuring-in-function-parameters/) instead:
+
+```js
+export default {
+	computed: {
+		d: ({ a, b, c }) => a = b + c
+	}
+};
+```
+
+The Svelte compiler can still see which values `d` depends on, but it's no longer injecting values — it just passes the component state object into each computed property.
+
+Again, you don't need to make this change manually — just run svelte-upgrade on your components, as shown above.
+
+
+## Sorry, IE11. It's not you, it's... well actually, yeah. It's you
+
+Svelte v1 was careful to only emit ES5 code, so that you wouldn't be forced to faff around with transpilers in order to use it. But it's 2018 now, and almost all browsers support modern JavaScript. By ditching the ES5 constraint, we can generate leaner code.
+
+If you need to support IE11 and friends, you will need to use a transpiler like [Babel](http://babeljs.io/repl) or [Bublé](http://buble.surge.sh/).
+
+
+## New lifecycle hooks
+
+In addition to `oncreate` and `ondestroy`, Svelte v2 adds two more [lifecycle hooks](guide#lifecycle-hooks) for responding to state changes:
+
+```js
+export default {
+	onstate({ changed, current, previous }) {
+		// this fires before oncreate, and
+		// whenever state changes
+	},
+
+	onupdate({ changed, current, previous }) {
+		// this fires after oncreate, and
+		// whenever the DOM has been updated
+		// following a state change
+	}
+};
+```
+
+You can also listen to those events programmatically:
+
+```js
+component.on('state', ({ changed, current, previous }) => {
+	// ...
+});
+```
+
+
+## component.observe
+
+With the new lifecycle hooks, we no longer need the `component.observe(...)` method:
+
+```js
+// before
+export default {
+	oncreate() {
+		this.observe('foo', foo => {
+			console.log(`foo is now ${foo}`);
+		});
+	}
+};
+
+// after
+export default {
+	onstate({ changed, current }) {
+		if (changed.foo) {
+			console.log(`foo is now ${current.foo}`);
+		}
+	}
+};
+```
+
+This shrinks the amount of code Svelte needs to generate, and gives you more flexibility. For example, it's now very easy to take action when any one of *several* properties have changed, such as the redrawing a canvas without debouncing several observers.
```

Comment:
This should be "such as the redrawing of a canvas" or "such as redrawing a canvas".